src/api/v1/store/index.js

```diff
@@ -0,0 +1,337 @@
+      res.json({ error: null, collection });
+    } catch (err) {
+      return next(err);
+    }
+  })
+
+  .patch(async (req, res, next) => {
+    if (!req.params.uuid || !req.body.collection) {
+      return next(new error.BadRequest('UUID and partial merchandise collection object must be provided'));
+    }
+
+    try {
+      let collection = await MerchandiseCollection.findByUUID(req.params.uuid);
+      if (!collection) return next(new error.BadRequest('No such merchandise collection found'));
+      collection = await collection.update(MerchandiseCollection.sanitize(req.body.collection));
+      res.json({ error: null, collection });
+    } catch (err) {
+      return next(err);
+    }
+  })
+
+  .delete(async (req, res, next) => {
+    if (!req.params.uuid) return next(new error.BadRequest('UUID must be provided'));
+
+    try {
+      const collection = await MerchandiseCollection.findByUUID(req.params.uuid);
+      if (!collection) return next(new error.BadRequest('No such merchandise collection found'));
+      const merchandise = await Merchandise.findAllByCollection(collection.uuid, req.user.isAdmin());
+      const itemsHaveBeenOrdered = await Promise.all(merchandise.map((m) => OrderItem.itemHasBeenOrdered(m.uuid)));
+      if (any(itemsHaveBeenOrdered)) {
+        return next(new error.BadRequest('An item from this collection has been previous ordered'));
+      }
+      await Promise
+        .all(merchandise.map((m) => Merchandise.destroyByUUID(m.uuid)))
+        .then(() => MerchandiseCollection.destroyByUUID(collection.uuid));
+      res.json({ error: null });
+    } catch (err) {
+      next(err);
+    }
+  });
+
+router.route('/merchandise/:uuid?')
+
+  /**
+   * Given a merch item UUID in the URI, return the matching item or null if no such item was found.
+   * Otherwise, returns all items listed on the store, including hidden items for admins.
+   */
+  .get(async (req, res, next) => {
+    if (!req.params.uuid) return next(new error.BadRequest('UUID must be provided'));
+
+    try {
+      const merchandise = await Merchandise.findByUUID(req.params.uuid)
+        .then((m) => (req.user.isAdmin() ? m : m.getPublicItem()));
+      res.json({ error: null, merchandise });
+    } catch (err) {
+      return next(err);
+    }
+  })
+
+  /**
+   * All further requests on this route require authentication and admin access.
+   */
+  .all((req, res, next) => {
+    if (!req.user.isAdmin()) return next(new error.Forbidden());
+    return next();
+  })
+
+  /**
+   * Creates an item, given a 'merchandise' object in the request body, and returns the created item
+   * upon success. Required fields: itemName, price, description. Optional fields: picture, quantity,
+   * discountPercentage, hidden. All other fields will be ignored.
+   */
+  .post(async (req, res, next) => {
+    if (!req.body.merchandise) return next(new error.BadRequest('Merchandise item must be provided'));
+
```

Comment:
POST `/store/merchandise/` throws an error if no collection uuid is provided or if collection is null in request body. Not all items need to belong to a collection though, right?

Reply:
Items must belong to a collection; it makes implementation (and merch taxonomy) much simpler and design team signed off on it.